pkg/cnab/config-adapter/stamp_test.go

```diff
@@ -29,6 +29,8 @@ func TestConfig_GenerateStamp(t *testing.T) {
 	require.NoError(t, err, "DigestManifest failed")
 	assert.Equal(t, simpleManifestDigest, stamp.ManifestDigest)
 	assert.Equal(t, map[string]MixinRecord{"exec": {}}, stamp.Mixins, "Stamp.Mixins was not populated properly")
+	assert.Equal(t, pkg.Version, stamp.Version)
```

Comment:
During unit tests, pkg.Version and Commit are empty strings by default. So even if you weren't setting these values on the stamp the test would pass. Here's what I recommend you do so that we can test that it's actually being set:

At the top of the test set pkg.Version/Commit to values and then use a defer to cleanup the change (those are package level variables and the change would persist for the next test).

```go
pkg.Version = "v1.2.3"
pkg.Commit = "abc123"
defer func() {
  pkg.Version = ""
  pkg.Commit = ""
}
```

Reply:
I'm not sure the value of the `pkg.Version` and `pkg.Commit` that I should assign at the beginning of the test. 

Assigning any kind of value in advance to the `pkg.Version` at the beginning of the test will make the result of `stamp.ManifestDigest` vary thus making the assertion of `assert.Equal(t, simpleManifestDigest, stamp.ManifestDigest)` always fails because it differs to the `simpleManifestDigest` that has been declared at the top of the test file, while the value of `pkg.Commit` does not change at all. 

![image](https://user-images.githubusercontent.com/7043511/107848307-031cda80-6e25-11eb-86b9-d148e6a36593.png)

Reply:
Sorry for the late reply!

I see what you mean, so why don't we add a new test based on that existing one that only checks if we set the version and commit on the stamp properly? That way we can avoid the problem you discovered, where the digest changes.

```go
func TestConfig_GenerateStamp_IncludeVersion(t *testing.T) {
	// Do not run this test in parallel
	// Still need to figure out what is introducing flakey-ness

	pkg.Version = "v1.2.3"
	pkg.Commit = "abc123"
	defer func() {
		pkg.Version = ""
		pkg.Commit = ""
	}()

	c := config.NewTestConfig(t)
	c.TestContext.AddTestFile("../../manifest/testdata/simple.porter.yaml", config.Name)

	m, err := manifest.LoadManifestFrom(c.Context, config.Name)
	require.NoError(t, err, "could not load manifest")

	a := NewManifestConverter(c.Context, m, nil, nil)
	stamp, err := a.GenerateStamp()
	require.NoError(t, err, "DigestManifest failed")
	assert.Equal(t, "v1.2.3", stamp.Version)
	assert.Equal(t, "abc123", stamp.Commit)
}
```